exporters/otlp/otlpgrpc/mock_collector_test.go

```diff
@@ -174,6 +177,19 @@ func (mc *mockCollector) GetMetrics() []*metricpb.Metric {
 	return mc.getMetrics()
 }
 
+// WaitForConn will wait indefintely for a connection to be estabilished
+// with the mockCollector before returning.
+func (mc *mockCollector) WaitForConn() {
+	for {
+		select {
+		case <-mc.ln.C:
+			return
+		default:
+			runtime.Gosched()
+		}
+	}
```

Comment:
Nitpick.  This logic seems more at home in the listener, and should just be a proxy call here.